aspnet/fundamentals/servers.rst

```diff
@@ -1,8 +1,274 @@
+
+ASP.NET 5 ships with support for 3 different servers:
+
+- Microsoft.AspNet.Loader.IIS (Helios)
+- Microsoft.AspNet.Server.WebListener (WebListener)
+- Microsoft.AspNet.Server.Kestrel (Kestrel)
+
+ASP.NET 5 does not directly listen for requests, but instead relies on the HTTP server implementation to surface the request to the application as a set of `feature interfaces`_ composed into an HttpContext. Both Helios and WebListener are Windows-only; Kestrel is designed to run cross-platform. You can configure your application to be hosted by any or all of these servers by specifying commands in your ``project.json`` file. You can even specify an application entry point for your application, and run it as an executable (using ``dnx . run``) rather than hosting it in a separate process.
+
+I've configured the sample project for this article to support each of these hosting options as separate commands in ``project.json``:
+
+.. literalinclude:: servers/sample/ServersDemo/src/ServersDemo/project.json
+	:lines: 1-17
+	:emphasize-lines: 12-14
+	:linenos:
+	:language: javascript
+	:caption: project.json (truncated)
+
+The ``run`` command will execute the application via its ``void main()`` method defined in ``program.cs``. In this case, this has been set up to configure and start an instance of ``WebListener``.
+
+.. literalinclude:: servers/sample/ServersDemo/src/ServersDemo/program.cs
+	:linenos:
+	:emphasize-lines: 27-35
+	:language: c#
+	:caption: program.cs
+
+
+Supported Features by Server
+----------------------------
+
+ASP.NET defines a number of :doc:`request-features` which may be supported on different server implementations. The following table lists the different features and the servers supporting them.
+
+.. list-table::
+	:header-rows: 1
+	
+	* - Feature
+	  - IIS / Helios
+	  - WebListener
+	  - Kestrel
+	* - IHttpRequestFeature
+	  - Yes
+	  - Yes
+	  - Yes
+	* - IHttpResponseFeature
+	  - Yes
+	  - Yes
+	  - Yes
+	* - IHttpAuthenticationFeature
+	  - Yes
+	  - Yes
+	  - No
+	* - IHttpUpgradeFeature
+	  - Yes
+	  - Yes
+	  - Yes
+	* - IHttpBufferingFeature
+	  - Yes
+	  - Yes
+	  - No
+	* - IHttpConnectionFeature
+	  - Yes
+	  - Yes
+	  - No
+	* - IHttpRequestLifetimeFeature
+	  - Yes
+	  - Yes
+	  - No
+	* - IHttpSendFileFeature
+	  - Yes
+	  - Yes
+	  - No
+	* - IHttpWebSocketFeature
+	  - Yes
+	  - Yes
+	  - No
```

Comment:
Can you elaborate? Should I say something like:
No*
* To add support for web sockets in Kestrel, use the WebSocketMiddleware